storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCenter.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.container.cgroup;
+
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class CgroupCenter implements CgroupOperation {
+
+    private static Logger LOG = LoggerFactory.getLogger(CgroupCenter.class);
+
+    private static CgroupCenter instance;
+
+    private CgroupCenter() {
+
+    }
+
+    public synchronized static CgroupCenter getInstance() {
+        if (instance == null) {
+            instance = new CgroupCenter();
+        }
+        return CgroupUtils.enabled() ? instance : null;
+    }
+
+    @Override
+    public List<Hierarchy> getHierarchies() {
+
+        Map<String, Hierarchy> hierarchies = new HashMap<String, Hierarchy>();
+
+        try (FileReader reader = new FileReader(CgroupUtils.MOUNT_STATUS_FILE);
+             BufferedReader br = new BufferedReader(reader)) {
+            String str = null;
+            while ((str = br.readLine()) != null) {
+                String[] strSplit = str.split(" ");
+                if (!strSplit[2].equals("cgroup")) {
+                    continue;
+                }
+                String name = strSplit[0];
+                String type = strSplit[3];
+                String dir = strSplit[1];
+                Hierarchy h = new Hierarchy(name, CgroupUtils.getSubSystemsFromString(type), dir);
+                hierarchies.put(type, h);
```

Comment:
"type" is just a comma delimited list of subsystems that are mounted on a hierarchy. Cgroup does not support mounting the same set of subsystems on multiple heirarchies (a.k.a mirrored hierarchies).  Thus, using the comma delimited list of subsystems as key is going to be distinct